constructor/osxpkg.py

```diff
@@ -423,6 +425,79 @@ def pkgbuild_prepare_installation(info):
         shutil.rmtree(f"{pkg}.expanded")
 
 
+def create_plugins(pages: list = None, codesigner: CodeSign = None):
+    if not pages:
+        return
+    elif isinstance(pages, str):
+        pages = [pages]
+
+    fresh_dir(PLUGINS_DIR)
+    xcodebuild = None
+
+    for page in pages:
+        xcodeproj_dirs = [
+            file.resolve()
+            for file in Path(page).iterdir()
+            if file.suffix == ".xcodeproj"
+        ]
+        if xcodeproj_dirs:
+            if not xcodebuild:
+                xcodebuild = shutil.which("xcodebuild")
+                if not xcodebuild:
+                    raise RuntimeError(
+                        "Plugin directory contains an uncompiled project,"
+                        " but xcodebuild is not available."
+                    )
+                try:
+                    subprocess.run([xcodebuild, "--help"], check=True, capture_output=True)
+                except subprocess.CalledSubprocessError:
+                    raise RuntimeError(
+                        "Plugin directory contains an uncomplied project,"
+                        " but xcodebuild requires XCode to compile plugins."
+                    )
+            for xcodeproj in xcodeproj_dirs:
+                build_cmd = [
+                    xcodebuild,
+                    "-project",
+                    str(xcodeproj),
+                    f"CONFIGURATION_BUILD_DIR={PLUGINS_DIR}",
+                    # do not create dSYM debug symbols directory
+                    "DEBUG_INFORMATION_FORMAT=",
+                ]
+                explained_check_call(build_cmd)
+        else:
+            plugin_name = os.path.basename(page)
+            page_in_plugins = join(PLUGINS_DIR, plugin_name)
+            shutil.copytree(page, page_in_plugins)
+
+    if codesigner:
+        with NamedTemporaryFile(suffix=".plist", delete=False) as entitlements:
+            plist = {
+                "com.apple.security.cs.allow-unsigned-executable-memory": True,
+                "com.apple.security.cs.disable-library-validation": True,
+            }
+            plist_dump(plist, entitlements)
```

Comment:
Since you implemented the plist dumper in the codesign class already, can we just pass the dict directly to the `.sign_bundle()` call below?

Reply:
This would create a new plist file for each plugin even though they are the same entitlements file.

An added benefit here is that both branches of the `sign_bundle()` call are covered with the same example test, but that's incidental.